arwen/ethapi/ethereumei.go

```diff
@@ -618,19 +619,19 @@ func ethrevert(context unsafe.Pointer, dataOffset int32, length int32) {
 	output := arwen.GetOutputContext(context)
 	metering := arwen.GetMeteringContext(context)
 
+	gasToUse := metering.GasSchedule().EthAPICost.Revert
+	gasToUse += metering.GasSchedule().BaseOperationCost.PersistPerByte * uint64(length)
+	metering.UseGas(gasToUse)
+
 	data, err := runtime.MemLoad(dataOffset, length)
-	if withFault(err, context) {
+	if arwen.WithFault(err, context, true) {
 		return
 	}
 
 	output.ClearReturnData()
 	output.Finish(data)
-	// TODO replace this with a breakpoint?
-	runtime.SignalUserError("revert")
 
-	gasToUse := metering.GasSchedule().EthAPICost.Revert
-	gasToUse += metering.GasSchedule().BaseOperationCost.PersistPerByte * uint64(length)
-	metering.UseGas(gasToUse)
+	runtime.SignalUserError("revert")
```

Comment:
`revert` seems correct. @sasurobert, ok?

Reply:
signal error can be called directly from the first line. All the data is thrown out in any case.

Reply:
Moving `SignalUserError()` above `WithFault()` will not change anything (they will override each other completely), but the function is more readable with `SignalUserError()` at the end.